src/gen/run_puzzle.rs

```diff
@@ -22,8 +23,14 @@ pub fn run_puzzle(
     max_cost: u64,
     flags: u32,
 ) -> Result<SpendBundleConditions, ValidationErr> {
-    let puzzle = node_from_bytes(a, puzzle)?;
-    let solution = node_from_bytes(a, solution)?;
+    let (puzzle, solution) = if (flags & ALLOW_BACKREFS) != 0 {
```

Comment:
Maybe same trick here, assign intermediate function `nfb` to either `node_from_bytes` or `node_from_bytes_backrefs` based on flags and use that.